pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -70,6 +191,22 @@ export function make_btrfs_subvolume_page(parent, volume, subvol) {
         });
     }
 
+    // If the current subvolume is mounted rw with an fstab entry or any parent
+    // subvolume is mounted rw with an fstab entry allow subvolume creation.
+    let create_excuse = "";
+    if (!mounted && !mount_point_in_parent)
+        create_excuse = _("One of the subvolumes parents needs to be mounted");
+    if (!mounted && subvol.id == 5)
+        create_excuse = _("Subvolume needs to be mounted");
+    if (mounted && opt_ro)
+        create_excuse = _("Subvolume needs to be mounted rw");
```

Comment:
Let's not say "rw" but "Subvolume needs to be writable".

Also, I think creation is possible if one of the parents is mounted rw, and get_mount_point_in_parent takes that into account.

So what about this:
```
if (!mount_point_in_parent) {
      if (!mounted)
        create_excuse = _("Subvolume needs to be mounted");
    if (!mounted && subvol.id == 5)
        create_excuse = _("Subvolume needs to be mounted");
    if (mounted && opt_ro)
        create_excuse = _("Subvolume needs to be mounted writable");
}
```

Thus, we only talk to the user in terms of the current subvolume and never about parents.

Reply:
Hmm I don't think this is true. `mount_point_in_parent` returns the mount_point for the current parent and doesn't become true. It breaks the current test_case for `mounted && opt_ro`.

```
/dev/sda on /run/ro type btrfs (ro,relatime,seclabel,ssd,space_cache=v2,subvolid=264,subvol=/ro)
```

Reply:
> It breaks the current test_case

There is a test case where we expect that creation inside the subvolume "ro" (mounted at "/run/ro") should be disabled, but with my proposed changes, it is enabled. Right?

This happens because the root subvol is still mounted read-write at "/run/butter" and creation of subvolumes of "ro" can be done inside the mount point "/run/butter/ro".

(I think, I'll check this now.)

Reply:
See my "PROPOSAL" commit.